source/views/sis/course-search/search.js

```diff
@@ -72,6 +72,18 @@ type State = {
 	query: string,
 }
 
+function keywordSearch(query: string, item: string) {
+	const keywords = query.split(' ')
+	const itemWords = item.split(' ')
+	return keywords.every(keyword =>
+		itemWords.some(
+			word =>
+				hamming(keyword, word.slice(0, keyword.length)) < 2 &&
+				keyword.length <= word.length,
```

Comment:
Just one optimization: let's move the length comparison to be first, so the easy check can short-circuit the more involved one.